File: DesignPatternsDotNet/StateDesignPattern/ChargingState.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace StateDesignPattern
{
  public class ChargingState : IState{
	private IMobile myMobile;
	public ChargingState(IMobile aMobile){
		myMobile = aMobile;
	}
    public void Charge()
    {
		Console.WriteLine("Mobile is connected to PC for charging only");
	}
    public void UsbMount()
    {
		Console.WriteLine("While Charging Only mode device can not mount as USB storage, please select USB mount from Device");
		myMobile.State=myMobile.GetUsb();
	}
    public void AdbConnect()
    {
		Console.WriteLine("While Charging Only mode we can not access ADB , please select Android Debugging from Device settings");
		myMobile.State=myMobile.GetAdb();
	}
}
}

File: DesignPatternsJava/src/main/java/com/stateDesignPattern/ChargingState.java
package com.stateDesignPattern;

public class ChargingState implements IState{
	private IMobile myMobile;
	public ChargingState(IMobile aMobile){
		myMobile = aMobile;
	}
	public void charge() {
		System.out.println("Mobile is connected to PC for charging only");
	}
	public void usbMount() {
		System.out.println("While Charging Only mode device can not mount as USB storage, please select USB mount from Device");
		myMobile.setMobileState(myMobile.getUsb());
	}
	public void adbConnect() {
		System.out.println("While Charging Only mode we can not access ADB , please select Android Debugging from Device settings");
		myMobile.setMobileState(myMobile.getAdb());
	}
}
